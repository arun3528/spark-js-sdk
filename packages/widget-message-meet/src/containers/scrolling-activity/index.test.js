import React from 'react';
import createComponentWithIntl from '../../utils/createComponentWithIntl';
import {Provider} from 'react-redux';

import store from '../../selectors/fixtures/mock-store';

import ScrollingActivity from '.';

store.indicators = {typing: []};

const onActivityDelete = jest.fn();
const onActivityFlag = jest.fn();

describe(`ScrollingActivity container`, () => {
  it(`renders properly`, () => {
    const component = createComponentWithIntl(
      <Provider store={store}>
        <ScrollingActivity
          onActivityDelete={onActivityDelete}
          onActivityFlag={onActivityFlag}
        />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it(`renders properly loading history`, () => {
    const component = createComponentWithIntl(
      <Provider store={store}>
        <ScrollingActivity
          isLoadingHistoryUp
          onActivityDelete={onActivityDelete}
          onActivityFlag={onActivityFlag}
        />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
