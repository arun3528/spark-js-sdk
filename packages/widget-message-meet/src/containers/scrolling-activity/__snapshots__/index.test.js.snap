exports[`ScrollingActivity container renders properly 1`] = `
<div
  className={undefined}
  onScroll={undefined}>
  <div>
    <div
      className="activity-list">
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-self" />
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Ricky Testerson
              </div>
              <div
                className="published">
                1/3/17, 12:22 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "hi",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Delete this message" />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="separator">
          <p
            className="separator-text">
            <span>
              January 6
            </span>
          </p>
        </div>
      </div>
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-letter">
                J
              </span>
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Jon Langley
              </div>
              <div
                className="published">
                1/6/17, 4:17 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "123",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action isHighlighted">
              <div
                className="post-action-item">
                <button
                  className="button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="action-spacer" />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="indicators">
      <div />
    </div>
  </div>
</div>
`;

exports[`ScrollingActivity container renders properly loading history 1`] = `
<div
  className={undefined}
  onScroll={undefined}>
  <div>
    <div
      className="spinner-container">
      <div
        className="spinner" />
    </div>
    <div
      className="activity-list">
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-self" />
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Ricky Testerson
              </div>
              <div
                className="published">
                1/3/17, 12:22 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "hi",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Delete this message" />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="separator">
          <p
            className="separator-text">
            <span>
              January 6
            </span>
          </p>
        </div>
      </div>
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-letter">
                J
              </span>
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Jon Langley
              </div>
              <div
                className="published">
                1/6/17, 4:17 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "123",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action isHighlighted">
              <div
                className="post-action-item">
                <button
                  className="button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="action-spacer" />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="indicators">
      <div />
    </div>
  </div>
</div>
`;
